### PR TITLE
Add changelog

### PR DESCRIPTION
**Pull request by clayreimann**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#Change Log
+
+## 3.0
+**Features**
+* Revised datastore interface 
+    * Removed hibernate-isms
+    * Made key-value persistence easier to support
+* New pre/post hook model
```
Comment (DeathByTape) [inline changelog.md]: Refer to this as `lifecycle` for consistency

File: changelog.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+* Updated audit logger interface
+* Removed all deprecated features
+    * SecurityMode
+    * `any` and `all` permission syntax
+    * Must use Elide.Builder
+* InMemoryDataStore is now the reference datastore implementation
+
+**Fixes**
+* Close transactions properly
```
Comment (DeathByTape) [inline changelog.md]: Maybe add that we upgrade _all_ deps? Or we will (working on now)? :)

Comment (aklish) [inline changelog.md]: Removed persistence store (Hibernate 5+ entity manager backed store)

File: changelog.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#Change Log
+
+## 3.0
+**Features**
+* Revised datastore interface 
+    * Removed hibernate-isms
+    * Made key-value persistence easier to support
+* New pre/post hook model
+* Updated audit logger interface
```
Comment (aklish) [inline changelog.md]: Is this true?

File: changelog.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    * Made key-value persistence easier to support
+* Changed lifecycle hook model
+* Updated audit logger interface
+* Removed all deprecated features, e.g.
+    * SecurityMode
+    * `any` and `all` permission syntax
+    * Required use of `ElideSettingsBuilder`
+    * Removed `PersistenceStore` from Hibernate 5 datastore
+* Made `InMemoryDataStore` the reference datastore implementation
+* Promoted `DefaultOpaqueUserFunction` to be a top-level class
```
Comment (DeathByTape) [inline changelog.md]: In the same vein, we should briefly mention our refactoring of the `Elide.Builder` -> top-level `ElideSettingsBuilder`

**Comment by DeathByTape** [comment]
👍

**Pull request merged**